src/Simulation/Simulation.jl

```diff
@@ -260,60 +269,66 @@ function Grid(sim::Simulation{T, Cylindrical};
-    important_r_points = merge_close_ticks(important_r_points)
-    important_r_points = initialize_axis_ticks(important_r_points; max_ratio = T(max_distance_ratio))
-    important_r_points = fill_up_ticks(important_r_points, max_distance_r)
-
-    append!(important_z_points, endpoints(world.intervals[3])...)
-    important_z_points = unique!(sort!(important_z_points))
+    iL = searchsortedfirst(important_r_ticks, world.intervals[1].left)
+    iR = searchsortedfirst(important_r_ticks, world.intervals[1].right)
+    important_r_ticks = unique(map(t -> isapprox(t, 0, atol = 1e-12) ? zero(T) : t, important_r_ticks[iL:iR]))
+    important_r_ticks = merge_close_ticks(important_r_ticks)
+    imp2order_r_ticks = merge_close_ticks(second_order_imp_ticks[1], min_diff = world_Δs[1] / 20)
+    important_r_ticks = merge_second_order_important_ticks(important_r_ticks, imp2order_r_ticks)   
+    important_r_ticks = initialize_axis_ticks(important_r_ticks; max_ratio = T(max_distance_ratio))
+    important_r_ticks = fill_up_ticks(important_r_ticks, max_distance_r)
+
+    append!(important_z_ticks, endpoints(world.intervals[3])...)
+    important_z_ticks = unique!(sort!(important_z_ticks))
     if add_points_between_important_point
-        important_z_points = sort!(vcat(important_z_points, StatsBase.midpoints(important_z_points)))
+        important_z_ticks = sort!(vcat(important_z_ticks, StatsBase.midpoints(important_z_ticks)))
     end
-    iL = searchsortedfirst(important_z_points, world.intervals[3].left)
-    iR = searchsortedfirst(important_z_points, world.intervals[3].right)
-    important_z_points = unique(map(t -> isapprox(t, 0, atol = 1e-12) ? zero(T) : t, important_z_points[iL:iR]))
-    important_z_points = merge_close_ticks(important_z_points)
-    important_z_points = initialize_axis_ticks(important_z_points; max_ratio = T(max_distance_ratio))
-    important_z_points = fill_up_ticks(important_z_points, max_distance_z)
-
-    append!(important_φ_points, endpoints(world_φ_int)...)
-    important_φ_points = unique!(sort!(important_φ_points))
+    iL = searchsortedfirst(important_z_ticks, world.intervals[3].left)
+    iR = searchsortedfirst(important_z_ticks, world.intervals[3].right)
+    important_z_ticks = unique(map(t -> isapprox(t, 0, atol = 1e-12) ? zero(T) : t, important_z_ticks[iL:iR]))
+    important_z_ticks = merge_close_ticks(important_z_ticks)
+    imp2order_z_ticks = merge_close_ticks(second_order_imp_ticks[3], min_diff = world_Δs[3] / 20)
+    important_z_ticks = merge_second_order_important_ticks(important_z_ticks, imp2order_z_ticks)
+    important_z_ticks = initialize_axis_ticks(important_z_ticks; max_ratio = T(max_distance_ratio))
+    important_z_ticks = fill_up_ticks(important_z_ticks, max_distance_z)
+
+    append!(important_φ_ticks, endpoints(world_φ_int)...)
+    important_φ_ticks = unique!(sort!(important_φ_ticks))
     if add_points_between_important_point
```

Comment:
Here again, it's `point` instead of `tick`.
Also, we use one function from `StatsBase` called `midpoints`.. 
Maybe `points` is a better choice than `ticks`?

Reply:
IMO ticks is better. In principle, both is fine. Just depends on your definition of a point 
(which also depends on the context).
Using ticks makes it a bit more clear what happens here (at least in my opinion).